clang/include/clang/3C/3CGlobalOptions.h

```diff
@@ -15,29 +15,61 @@
 
 #include "llvm/Support/CommandLine.h"
 
-extern bool Verbose;
-extern bool DumpIntermediate;
-extern std::string OutputPostfix;
-extern std::string OutputDir;
-extern bool HandleVARARGS;
-extern bool SeperateMultipleFuncDecls;
-extern bool EnablePropThruIType;
-extern bool ConsiderAllocUnsafe;
-extern bool AllTypes;
-extern bool NewSolver;
-extern std::string BaseDir;
-extern std::vector<std::string> AllocatorFunctions;
-extern bool AddCheckedRegions;
-extern bool WarnRootCause;
-extern bool WarnAllRootCause;
-extern bool DumpUnwritableChanges;
-extern bool AllowUnwritableChanges;
-extern bool AllowRewriteFailures;
-extern bool ItypesForExtern;
+// Options used to initialize 3C tool.
+//
+// See clang/docs/checkedc/3C/clang-tidy.md#_3c-name-prefix
+// NOLINTNEXTLINE(readability-identifier-naming)
+struct _3COptions {
+  bool DumpIntermediate;
+
+  bool Verbose;
+
+  std::string OutputPostfix;
+  std::string OutputDir;
+
+  std::string ConstraintOutputJson;
+
+  bool DumpStats;
+
+  std::string StatsOutputJson;
+
+  std::string WildPtrInfoJson;
+
+  std::string PerWildPtrInfoJson;
+
+  std::vector<std::string> AllocatorFunctions;
+
+  bool HandleVARARGS;
+
+  bool EnablePropThruIType;
+
+  std::string BaseDir;
+  bool AllowSourcesOutsideBaseDir;
+
+  bool AllTypes;
+
+  bool AddCheckedRegions;
+
+  bool EnableCCTypeChecker;
+
+  bool WarnRootCause;
+
+  bool WarnAllRootCause;
 
 #ifdef FIVE_C
-extern bool RemoveItypes;
-extern bool ForceItypes;
+  bool RemoveItypes;
+  bool ForceItypes;
 #endif
 
+  bool DumpUnwritableChanges;
+  bool AllowUnwritableChanges;
+
+  bool AllowRewriteFailures;
+
+  bool ItypesForExtern;
+};
+
+// NOLINTNEXTLINE(readability-identifier-naming)
+extern struct _3COptions _3CGlobalOptions;
```

Comment:
`_3CGlobalOptions` is long enough that I find it adds significant clutter to the code that references it.  Maybe consider a shorter name like `_3COpts`?

Reply:
Yes, I also noticed the extra clutter. I wanted the name to emphasize that this is a global state. I felt that a part of the issue with the old global variables was the it wasn't obvious to someone unfamiliar with the code that a variable was globally scoped.

I might shorten to something like `_3CGlobalOpts`, or maybe `Global3COpts` to get rid of the underscore.

Reply:
> I felt that a part of the issue with the old global variables was that it wasn't obvious to someone unfamiliar with the code that a variable was globally scoped.

Right, that was a major problem when we had 20+ different variables with completely different names.  If we have only one struct, it shouldn't be as hard for readers to learn once that it is global.  So I don't think the `Global` in the name is worth the clutter, but if you disagree, fine.